scripts/nb-tester/README.md

```diff
@@ -0,0 +1,168 @@
+# Qiskit documentation notebook tester
+
+A tool to execute notebooks for testing, with certain features useful for
+notebooks that run jobs on IBM Quantum backends.
+
+## Basic usage
+
+To use this tool, just run the `test-docs-notebooks` command, followed by a
+list of notebooks you'd like to execute.
+
+```
+test-docs-notebooks path/to/notebook.ipynb path/to/another.ipynb
+```
+
+* To write the executed notebooks to disk, include the `--write` argument.
+* To set a time limit for each cell, include the `--cell-timeout` argument,
+  followed by the time in seconds.
+
+## Patches
+
+A key feature of this tool is tricking notebooks into running jobs on a
+specific backend so we can test them without waiting in a queue. We do this by
+monkey-patching `QiskitRuntimeService.least_busy` to return the backend of our
+choosing. This technique requires notebooks always use `least_busy` to select a
+backend.
+
+A patch is just Python code that we run in the kernel before executing your
+notebook's code. You can provide your own patch code, or use one of our
+built-in patches.
+
+You specify patch information as a TOML dict string. This dict _must_ inculde a
+`patch` key, which is either a filepath to your custom patch, or the name of
+one of our built-in patches (choose from `qiskit-fake-provider`,
+`qiskit-ibm-runtime`, or `qiskit-fake-provider`).
+
+```
+{ patch="patch-name" }
+```
+
+To make them more flexible, the patch code can include variables inside curly
+braces. If a patch includes variables, you must provide them as extra arguments
+when specifying the patch. These arguments will be injected into your patch
+code using `.format(**args)`. For example, here's a simplified version of our
+built-in `qiskit-fake-provider` patch:
+
+```python
+# Example patch file
+from qiskit.providers.fake_provider import GenericBackendV2
+from qiskit_ibm_runtime import QiskitRuntimeService
+QiskitRuntimeService.least_busy = lambda *_: GenericBackendV2(num_qubits={num_qubits})
+```
+
+Note the variable `{num_qubits}`. To use this patch, include the values of any
+variables as extra arguments in the TOML dict.
+
+```
+{ patch="qiskit-fake-provider", num_qubits=3 }
+```
+
+There are two ways to provide patches: Through the CLI or through a config file.
+You cannot use both at the same time.
+
+### Set patches through the CLI
```

Comment:
Not sure why, but looks like this line isn't rendered properly

Reply:
Weird, must have somehow typed a different unicode space character 😂 
I re-typed it into GitHub and that fixed it (bce4f675a2311a9a2c4738b0d01c8a9f7a6c9314)